docs/proposals/15-manage-bastions-and-ssh-key-pair-rotation.md

```diff
@@ -0,0 +1,212 @@
+- Reuse of temporarily created necessary infrastructure resources by different users
+
+## Proposal
+
+### Involved Components
+The following is a list of involved components, that either need to be newly introduced or extended if already existing
+- `gardenctlv2` (or any other client)
+  - Creates `Bastion` resource in the garden cluster, see [resource example](#resource-example) below
+  - Establishes an SSH connection to a shoot node, using a bastion host as proxy
+  - Heartbeats / keeps alive the `Bastion` resource during SSH connection
+- Gardener extension provider <infra>
+  - Provider specific bastion controller
+  - Should be added to gardener-extension-provider-<infra> repos, e.g. https://github.com/gardener/gardener-extension-provider-aws/tree/master/pkg/controller
+  - Has the permission to update the `Bastion/status` subresource on the seed cluster
+  - Runs on seed (of course)
+- Gardener API Server (`GAPI`)
+  - New `operations.gardener.cloud` API Group
+  - New resource type `Bastion`
+  - New Admission Webhooks for `Bastion` resource
+  - `SeedAuthorizer`: The `SeedAuthorizer` and dependency graph needs to be extended to consider the `Bastion` resource https://github.com/gardener/gardener/tree/master/pkg/admissioncontroller/webhooks/auth/seed/graph
+- Gardener Controller Manager (`GCM`)
+  - `Bastion` heartbeat controller
+    - Cleans up `Bastion` resource on missing heartbeat.
+    - Is configured with a `maxLifetime` and `timeToLife` for the `Bastion` resource
+- `gardenlet`
+  - Similar to `BackupBucket`s or `BackupEntry`, the `gardenlet` watches the `Bastion` resource in the garden cluster and creates a seed-local `Bastion` resource, on which the provider specific bastion controller acts upon
+- Gardener (RBAC)
+  - The project `admin` role should be extended to allow CRUD operations on the `Bastion` resource. The `gardener.cloud:system:project-member-aggregation` `ClusterRole` needs to be updated accordingly (https://github.com/gardener/gardener/blob/master/charts/gardener/controlplane/charts/application/templates/rbac-user.yaml)
+
+### SSH Flow
+0. Users should only get the RBAC permission to `create` / `update` `Bastion` resources for a namespace, if they should be allowed to SSH onto the shoot nodes in this namespace. A project member with `admin` role will have these permissions.
+1. User/`gardenctlv2` creates `Bastion` resource in garden cluster (see [resource example](#resource-example) below)
+    - First, gardenctl would figure out the external IP. Either by calling an external service (gardenctl (v1) uses https://github.com/gardener/gardenctl/blob/master/pkg/cmd/miscellaneous.go#L226) or by calling a binary that prints the external IP(s) to stdout. The binary should be configurable. The result is set under `spec.clientIP`
+    - the public PGP key of the user is set under `spec.pgpPublicKey`. The key that should be used needs to be configured beforehand by the user
```

Comment:
Can you elaborate why we don't want the user to configure his public RSA key for SSH access but instead go an extra mile via PGP?

Reply:
@vlerenc reasoned about it here in the old proposal https://github.com/gardener/gardenctl/issues/508#issuecomment-803562759, or do you want me to motivate it in the GEP?

Reply:
Okay, thanks for pointing that out. I think it would at least help to have this motivation as a reference for future design decisions.

Reply:
@timuthy I motivated it here https://github.com/gardener/gardener/pull/3802/commits/a0609293e43cdc4a4a02fc67546bae6930b41bb7
@vlerenc basically I copied your quote from https://github.com/gardener/gardenctl/issues/508#issuecomment-803562759, I hope you are ok with that?